### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by cpthftg**
Basic scaffolding for dashboard, UI only so far, no logic implemented.

### CODE REVIEW

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -5,10 +5,64 @@
...
+use App\Models\Task;
+use App\Models\Event;
+use Illuminate\Http\Request;
 
-class DashboardController extends Controller
-{
+class DashboardController extends Controller {
     public function index() {
-        return Inertia::render('Dashboard');
+        $user_id = User::query()->where('id', '=', 1)->get()[0]->id;
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: Du kannst dein User, der den Request macht mithilfe dieses Code snippets erreichen:

```
public function index(Request $request) {
        $user = $request->user();
```

File: app/Http/Controllers/DashboardController.php [hunk 2]
```diff
@@ -5,10 +5,64 @@
...
+        $modules = Module::query()->where('user_id', '=', $user_id)->get();
+        $tasks = Task::query()->where('user_id', '=', $user_id)->get();
+        $events = Event::query()
+            ->where('user_id', $user_id)
+            ->whereDate('start', '>=', now()->toDateString())
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: Würde den Zeitraum auf 1 Woche erweitern.

Zusätzlich sollte man schauen ob es laufende events gibt, also start in der Vergangenheit / ende in der Zukunft. Z.b.: wenn man sich eine Einführungswoche einträgt als Mehrtagesevent.

File: app/Http/Controllers/DashboardController.php [hunk 3]
```diff
@@ -5,10 +5,64 @@
...
+            ->get();
+
+        // Get all module ID's in one variable;
+        // Also count the amount of modules done
+        $module_ids = [];
+        $modules_done = 0;
+        foreach($modules as $module){
+            $module_ids[] = $module->id;
+
+            if ($module->status >= 100){
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: status will be an enum value. Please see:
https://github.com/f0reclone/wbh-lernplattform/pull/9/files#diff-6dcb0a44a32c5eda55a5714c1f27e95d05852427207d4ae4bfb6e685db4b724a

File: app/Http/Controllers/DashboardController.php [hunk 4]
```diff
@@ -5,10 +5,64 @@
...
+        $grade_sum = 0;
+        foreach($exams as $exam) {
+            $credit_points_total += $exam->credit_points;
+            $grade_sum += $exam->grade;
+
+            if ($exam->grade <= 4) {
+                $credit_points_achieved += $exam->credit_points;
+            }
+        }
+        $grade_average = $grade_sum / count($exams);
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: Division by 0 if no exams available.

Comment (f0reclone) [reply app/Http/Controllers/DashboardController.php]: Grades should also be weighted by number of credit points.

File: app/Http/Controllers/DashboardController.php [hunk 5]
```diff
@@ -5,10 +5,64 @@
...
+
+        // Sum the amount of credit points over all exams and count those that are done
+        // An exam is considered done if it has been graded with a passing grade <= 4
+        // Also calculate the average grade over all exams
+        $credit_points_total = 0;
+        $credit_points_achieved = 0;
+        $grade_sum = 0;
+        foreach($exams as $exam) {
+            $credit_points_total += $exam->credit_points;
+            $grade_sum += $exam->grade;
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: If an exam was not passed, the grade should not count. Also we should skip exams which are only planned yet.

File: app/Http/Controllers/DashboardController.php [hunk 6]
```diff
@@ -5,10 +5,84 @@
...
+            ->where('user_id', '=', $user_id)
+            ->get();
+        $tasks = Task::query()
+            ->where('user_id', '=', $user_id)
+            ->get();
+        $events = Event::query()
+            ->where('user_id', $user_id)
+            ->where(function ($query) {
+                // Events starting within the next seven days
+                $query->whereDate('start', '>=', now()->toDateString())
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: Needs to be properly wrapped.

```
->where(function ($query) {
  ->orWhere(function ($query) {
  // Events starting within the next 
  })
  ->orWhere(function ($query) {
  // OR Events that started in the past and will end within the next seven daysseven days
  });
});
```

File: app/Http/Controllers/DashboardController.php [hunk 7]
```diff
@@ -5,10 +5,84 @@
...
+        $exams = Exam::query()->whereIn('module_id', $module_ids)->get();
+
+        // Sum the amount of credit points over all exams and count those that are done
+        // An exam is considered done if it has been graded with a passing grade <= 4
+        // Also calculate the average grade over all exams, weighted by credit points
+        $credit_points_total = 0;
+        $credit_points_achieved = 0;
+        $grade_sum = 0;
+        foreach($exams as $exam) {
+            if ($exam->grade <= 4) {
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: The grade is saved as `40`.

File: app/Http/Controllers/DashboardController.php [hunk 8]
```diff
@@ -5,10 +5,84 @@
...
+                $credit_points_achieved += $exam->credit_points;
+                $credit_points_total += $exam->credit_points;
+                $grade_sum += $exam->grade * $exam->credit_points;
+            }
+        }
+
+        $grade_average = 0;
+        if (count($exams) != 0){
+            $grade_average = $grade_sum / count($exams);
```
Comment (f0reclone) [inline app/Http/Controllers/DashboardController.php]: Here we need to use the credit points achieved.

File: resources/js/Pages/Dashboard.vue [hunk 9]
```diff
@@ -1,25 +1,237 @@
...
-        </div>
-    </AuthenticatedLayout>
+		<template #header>
+			<h1 class="font-semibold text-xl text-gray-800 leading-tight">Übersicht</h1>
+		</template>
+		<!-- 0 - This is the outermost Div -->
+		<div class="py-6 flex justify-center items-center">
+
+			<!-- 1 - This is the second outermost Div-->
+			<div class="max-w-7xl mx-auto space-y-6 h-screen">
```
Comment (floh007) [inline resources/js/Pages/Dashboard.vue]: Remove 'h-screen' for the flex container that sets the background to extend if needed

**Pull request merged**